### PR TITLE
[improvement](pipeline) Support sharing hash table for broadcast join

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes

1. FE chooses one instance to build hash table.
2. Add `EmptySourceOperator` to adapt for `HashJoinBuildSink` which doesn't need to build hash table.

## Problem Summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.05 seconds
 load time: 471 seconds
 storage size: 17121632496 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104183725_clickbench_pr_73704.html

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -1109,4 +1114,32 @@ Status FragmentMgr::merge_filter(const PMergeFilterRequest* request,
...
+        return;
+    }
+    for (auto& node : params.fragment.plan.nodes) {
+        if (node.node_type != TPlanNodeType::HASH_JOIN_NODE ||
+            !node.hash_join_node.__isset.is_broadcast_join ||
+            !node.hash_join_node.is_broadcast_join) {
+            continue;
+        }
+
+        std::lock_guard<std::mutex> lock(_lock_for_shared_hash_table);
```
Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:1136]: why here need a lock here ? rethink the lock is necessary?

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**